### PR TITLE
Ensure/allow callers to specify iden for adding triggers and dmons (SYN-5334)

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/cortex.py [hunk 1]
```diff
@@ -4742,8 +4742,8 @@ async def addStormDmon(self, ddef):
         '''
         Add a storm dmon task.
         '''
-        iden = s_common.guid()
-        ddef['iden'] = iden
+        if ddef.get('iden') is None:
+            ddef['iden'] = s_common.guid()
```
Comment (vEpiphyte) [inline synapse/cortex.py:4746]: We should raise a s_exc.DupIden exception in the top handler here, similar to how `addCronJob(...)` works.

File: synapse/lib/stormtypes.py [hunk 2]
```diff
@@ -644,53 +644,54 @@ class LibDmon(Lib):
          'type': {'type': 'function', '_funcname': '_libDmonAdd',
                   'args': (
                       {'name': 'text', 'type': ['str', 'storm:query'],
-                       'desc': 'The Storm query to execute in the Dmon loop.', },
-                      {'name': 'name', 'type': 'str', 'desc': 'The name of the Dmon.', 'default': 'noname', },
+                       'desc': 'The Storm query to execute in the Dmon loop.'},
+                      {'name': 'name', 'type': 'str', 'desc': 'The name of the Dmon.', 'default': 'noname'},
+                      {'name': 'ddef', 'type': 'dict', 'desc': 'Additional daemon definition fields. ', 'default': None},
```
Comment (vEpiphyte) [inline synapse/lib/stormtypes.py:649]: We should probably add a Dmon schema that is checked somewhere.

Comment (vEpiphyte) [reply synapse/lib/stormtypes.py:649]: schema addition should be checked in the top half of the dmon handler

Comment (invisig0th) [reply synapse/lib/stormtypes.py:649]: schema exists and is checked by the bottom half of the handler ( which would need to be the case anyway ). Top half checking is mostly for avoiding the nexus log entry which in this case is pretty minor, but feel free to add a ticket about it. :+1:

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**